src/main/webapp/assets/js/index.js

```diff
@@ -77,32 +72,43 @@ function filterCategory(category) {
   }
 }
 
+/**
+ * Search businesses names based on search term.
+ */
+function searchBusinesses(business, searchTerms) {
+  let filtered = new Set();
+  for (let i = 0; i < searchTerms.length; i++) {
+     filtered.add(business.filter(dict => dict.data.name.toLowerCase().includes(searchTerms[i].toLowerCase())));
+  }
+  return filtered;
+}
+
 /**
  * Prevents user from entering an empty search term.
  */
 async function handleSearch() {
-  let searchTerm = document.getElementById("query").value;
+  let searchTerms = document.getElementById("query").value;
 
-  if (searchTerm == "") {
+  if (searchTerms == "") {
     alert("Enter a valid search term!");
     return false;
   }
 
-  let param = {'s': searchTerm}
-  let searchParams = new URLSearchParams(param).toString();
-
-  let searchData = await fetchUrlData(`/search?${searchParams}`);
+  // uses the initial businesses data retrieved when the page is loaded
+  let searchData = searchBusinesses(businesses, searchTerms.split(" "));
 
   if (searchData.length === 0) {
     alert("No search result found!");
     return false;
   } else {
-    document.getElementById("product-listings-title").innerText = `Search Result for "${searchTerm}": `;
+    document.getElementById("product-listings-title").innerText = `Search Result for "${searchTerms}": `;
     productListings = document.getElementById("product-listings");
     productListings.innerHTML = '';
 
-    for (let i = 0; i < searchData.length; i++) {
-      productListings.appendChild(homePageListingTemplate(searchData[i]));
```

Comment:
Consider changing "homePageListingTemplate" to a more appropriateName, i.e. createProductListing